c7n/resources/account.py

```diff
@@ -314,7 +353,7 @@ def process(self, resources, event=None):
                 filter_matched = None
                 if metric_filters_log_group:
                     for f in metric_filters_log_group:
-                        if f['filterPattern'] == self.data.get('log-metric-filter-pattern'):
+                        if self.op(f['filterPattern'], pattern):
```

Comment:
Assuming we've built the value filter above, we could use it here.

```suggestion
                        if vf(f):
```